src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,36 @@
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        {
+            var optionalIfElements = page.Elements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIfValue) 
+                || !string.IsNullOrEmpty(element.Properties.OptionalIfNotValue));
```

Comment:
If you always use "OptionalIfQuestionId" property, why not use that to select this element? If my understanding is right on how to use this feature... I am only guessing you would always need some previous questionId to run a check on.